torchmetrics/metric.py

```diff
@@ -66,6 +66,8 @@ class Metric(nn.Module, ABC):
         dist_sync_fn:
             Callback that performs the allgather operation on the metric state. When `None`, DDP
             will be used to perform the allgather.
+        should_sync_state_dict: Whether states should be synchronized while creating a checkpoint.
+            This would be part of the checkpoint and re-used when reloading the metric state_dict.
```

Comment:
`should_sync_state_dict` does not follow the same naming convention as `compute_on_step` and `dist_sync_on_step`.
Does torchmetrics want to have a consistent naming?
RFC @PyTorchLightning/core-metrics

Reply:
Great catch ! Changed to `dist_sync_state_dict`.